File: app/llms/solid/golang/ocp.go
package main

func CalculateDiscount(t string, price float64) float64 {
	if t == "student" {
		return price * 0.9
	}
	return price
}

type Discount interface {
	Apply(price float64) float64
}

type StudentDiscount struct{}

func (d StudentDiscount) Apply(price float64) float64 {
	return price * 0.9
}

File: app/llms/solid/javascript/ocp.js

function calculateDiscount(type, price) {
  if (type === "student") return price * 0.9;
  if (type === "senior") return price * 0.8;
  return price;
}

class Discount {
  apply(price) {
    return price;
  }
}

class StudentDiscount extends Discount {
  apply(price) {
    return price * 0.9;
  }
}

class SeniorDiscount extends Discount {
  apply(price) {
    return price * 0.8;
  }
}
